tests/test_main_loop.py

```diff
@@ -77,3 +64,41 @@ def do_test(with_serialization):
 
     do_test(False)
     do_test(True)
+
+
+def test_training_interrupt():
+    def process_batch(batch):
+        time.sleep(0.1)
+
+    algorithm = MockAlgorithm()
+    algorithm.process_batch = process_batch
+
+    main_loop = MockMainLoop(
+        algorithm=algorithm,
+        data_stream=IterableDataset(count()).get_example_stream(),
+        extensions=[Printing()]
+    )
+
+    p = Process(target=main_loop.run)
+    p.start()
+    time.sleep(0.1)
+    os.kill(p.pid, signal.SIGINT)
+    time.sleep(0.1)
+    assert p.is_alive()
+    os.kill(p.pid, signal.SIGINT)
+    time.sleep(0.2)
+    assert not p.is_alive()
+    p.join()
+
+
+def test_error():
+    ext = TrainingExtension()
+    ext.after_batch = MagicMock(side_effect=KeyError)
+    ext.on_error = MagicMock()
+    main_loop = MockMainLoop(extensions=[ext, FinishAfter(after_epoch=True)])
+    assert_raises(KeyError, main_loop.run)
+    ext.on_error.assert_called_once_with()
```

Comment:
It's possible to check that we have a log record `got_exception` here, isn't it?